static/src/javascripts/lib/storage.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+/* eslint consistent-return: 0, no-plusplus: 0 */
+
+class Storage {
+    storage: any;
+    available: ?boolean;
+
+    constructor(type: string) {
+        this.storage = window[type];
+        this.available = undefined;
+    }
+
+    isAvailable(): ?boolean {
```

Comment:
Unfortunately Opera Mini is still an issue - not localStorage support at all. If we don't guard the calls, it would throw Errors. There is a polyfill in pf.io, but it's only for localStorage (not sessionStorage) and it's not persistant.